lib/reverse_string.rb

```diff
@@ -1,5 +1,30 @@
 def reverse_characters(str)
+
+    i = 0    
+    n = str.size - 1
+    ansstr = " "*n
+    
+while(i<str.size)
```

Comment:
Indentation is a concern here. The block on lines 7-10 should all be indented a space, so that lines 7 and 10 are in line with the code on lines 3-5. Line 12 should also be indented.